File: transcript.php
<?php

$file = $argv[1];
$data = json_decode(file_get_contents($file));
$basename = basename($file);

// Load up speaker labels.
$labels = $data->results->speaker_labels->segments;
$speaker_start_times = [];
foreach ($labels as $label) {
  foreach ($label->items as $item) {
    $speaker_start_times[number_format($item->start_time, 3)] = $label->speaker_label;
  }
}

// Now we iterate through items and build the transcript
$items = $data->results->items;
$lines = [];
$line = '';
$speaker = NULL;
foreach ($items as $item) {
  $content = $item->alternatives[0]->content;
  if (property_exists($item, 'start_time')) {
    $current_speaker = $speaker_start_times[number_format($item->start_time, 3)];
  }
  elseif ($item->type == 'punctuation') {
    $line .= $content;
  }
  if ($current_speaker != $speaker) {
    if ($speaker) {
      $lines[] = [
        'speaker' => $speaker,
        'line' => $line,
      ];
    }
    $line = $content;
    $speaker = $current_speaker;
  }
  elseif ($item->type != 'punctuation') {
    $line .= ' ' . $content;
  }
}
// Record the last line since there was no speaker change.
$lines[] = [
  'speaker' => $speaker,
  'line' => $line,
];


// Finally, let's print out our transcript.
$fh = fopen($file . '-transcript.txt', 'w');
foreach ($lines as $line_data) {
  $line = $line_data['speaker'] . ': ' . $line_data['line'];
  fputs($fh, $line . "\n\n");
}
fclose($fh);

/* End of the transcript.php file */

File: transcript.py
#!/usr/bin/env python3
def main():
	import sys
	import json
	import datetime
	
	filename=sys.argv[1]
	print ("Filename: ", filename)
	with open(filename+'.txt','w') as w:
		with open(filename) as f:
			data=json.loads(f.read())
			labels = data['results']['speaker_labels']['segments']
			speaker_start_times={}
			for label in labels:
				for item in label['items']:
					speaker_start_times[item['start_time']] =item['speaker_label']
			items = data['results']['items']
			lines=[]
			line=''
			time=0
			speaker='null'
			i=0
			for item in items:
				i=i+1
				content = item['alternatives'][0]['content']
				if item.get('start_time'):
					current_speaker=speaker_start_times[item['start_time']]
				elif item['type'] == 'punctuation':
					line = line+content
				if current_speaker != speaker:
					if speaker:
						lines.append({'speaker':speaker, 'line':line, 'time':time})
					line=content
					speaker=current_speaker
					time=item['start_time']
				elif item['type'] != 'punctuation':
					line = line + ' ' + content
			lines.append({'speaker':speaker, 'line':line,'time':time})
			sorted_lines = sorted(lines,key=lambda k: float(k['time']))
			for line_data in sorted_lines:
				line='[' + str(datetime.timedelta(seconds=int(round(float(line_data['time']))))) + '] ' + line_data.get('speaker') + ': ' + line_data.get('line')
				w.write(line + '\n\n')


if __name__ == '__main__':
	main()
